e2e-product-testing/cypress/integration/common/pages.ts

```diff
@@ -0,0 +1,9 @@
+import { Given } from "cypress-cucumber-preprocessor/steps";
+
+Given("I am on a main page", () => {
+  cy.visit("https://marxan.vercel.app/", {
```

Comment:
I'm personally not a fan on relying e2e tests that are triggered as part of a CI on an instance deployed somewhere else
Also, if FE (or BE) adds a new e2e test, that would fail until a new develop is deployed (which is happening manually right know I think), so I think is not a good option (yet) to have it as a CI stage

Reply:
This is a typical black-box testing of deployed product as a whole.

It is a "golden target" to write requirements in this form and complete feature (merge it) when its fully done. We may not require this test to be present on develop/production until feature is ready (or mark it as "can fail").

What we had in mind is to ensure that most business-critical features *works*. Writing such for future/upcoming features is a different story.

Reply:
`This is a typical black-box testing of deployed product as a whole.`

The project every time gets built is the product as a whole.

I just think that since this is a part of the project, it would make more sense to be targeted to the project itself, as part of a CI stage. Otherwise I think it would make more sense having a standalone project that runs against a deployed instance, and just take care of having them in sync.

Very personal opinion tho

Reply:
I think we could start with what we have, and adjust as we change our deployment workflow.

We should actually be quite close to being able to run the whole product on GH Actions (https://github.com/Vizzuality/marxan-cloud/pull/118, which I need to resurrect from the recent mass-closure of PRs), at which stage we could target the local (GH Actions) app.

If we want to stick with the initial idea of running these tests against a staging environment, though, that would still be separate from GH Actions, and I would still like to use GH Actions to run these tests to avoid having to set up a runner for these Cypress tests...

I am not sure how much "full seed data" may make sense for running product tests (probably we can avoid, but who knows what we may want to test at some point soon), in which case we would not be able to populate ephemeral test instances such as those on GH Actions runners with full data as this requires lots of time - just as a side note.

Reply:
I am not sure about any strong value of testing "local" setup with product test - we want to test the real infrastructure (mostly) - with all the downsizes and exactly how user will experience it. (thus long timeouts are a nice indicator that page works slowly from UX perspective).

We can have them as an addition, of course.

> GH Actions runners with full data as this requires lots of time
we can prepare ready docker images with all the data already seeded and strong cache ; )